### PR TITLE
reduce travis-ci time of build_doc

### LINKED ISSUE #3303: build_doc in travis CI is slow
image
It always costs more than half an hour to run build_doc in travis CI.

### PR DESCRIPTION
**Pull request by luotao1**
fix #3303 
related #1854
1. build_doc don't depend on `import py_paddle`  now, thus, remove `import py_paddle` in both conf.py.cn.in and conf.py.en.in
2. As build_doc depends on `import paddle`, we can use `'@PROJ_ROOT@/python/paddle` instead of installing paddle python package, which will reduce a lot of time.

build_doc in Dockerfile ([build.sh](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/scripts/docker/build.sh)) will be updated in next PR.

### CODE REVIEW

File: paddle/scripts/travis/build_doc.sh [hunk 1]
```diff
@@ -35,6 +28,7 @@ TARGET_BRANCH="gh-pages"
 SOURCE_BRANCH="master"
 
 # Clone the repo to output directory
+mkdir output
```
Comment (wangkuiyi) [inline paddle/scripts/travis/build_doc.sh]: Should we use `rm -rf output` here?  It seems that `rmdir output` works only when output is an empty directory.

File: paddle/scripts/travis/build_doc.sh [hunk 2]
```diff
@@ -5,13 +5,6 @@ set -e
 mkdir -p $TRAVIS_BUILD_DIR/build
 cd $TRAVIS_BUILD_DIR/build
 
-# Compile paddle binaries first
-cmake .. -DCMAKE_BUILD_TYPE=Debug -DWITH_GPU=OFF -DWITH_DOC=OFF -DWITH_MKLDNN=OFF -DWITH_MKLML=OFF -DWITH_GOLANG=ON -DWITH_STYLE_CHECK=OFF
```
Comment (wangkuiyi) [inline paddle/scripts/travis/build_doc.sh]: Wow. Does this mean that the `build_doc` Travis job doesn't build Paddle twice now?

**Comment by wangkuiyi** [comment]
Travis CI's build_doc jobs failed:

```
Generating sphinx documentation: html
Running Sphinx v1.5.6
making output directory...
Traceback (most recent call last):
  File "/home/travis/build/PaddlePaddle/Paddle/python/paddle/trainer/config_parser.py", line 85, in <module>
    from paddle.proto.DataConfig_pb2 import DataConfig
  File "/home/travis/build/PaddlePaddle/Paddle/python/paddle/proto/__init__.py", line 15, in <module>
    from paddle.proto.TrainerConfig_pb2 import OptimizationConfig, TrainerConfig
ImportError: No module named TrainerConfig_pb2
Configuration error:
The configuration file (or one of the modules it imports) called sys.exit()
make[3]: *** [doc/CMakeFiles/paddle_docs] Error 1
make[2]: *** [doc/CMakeFiles/paddle_docs.dir/all] Error 2
make[1]: *** [doc/CMakeFiles/paddle_docs.dir/rule] Error 2
make: *** [paddle_docs] Error 2
```

I tried to help fixing it, but no idea how to do it.

**Comment by Xreki** [comment]
There are some other places that import py_paddle, and I'm afraid that building of py_paddle cannot be removed. See as following:
```bash
$ grep -r py_paddle *
paddle/v2/data_feeder.py:from py_paddle import DataProviderConverter
paddle/v2/data_feeder.py:        :type argument: py_paddle.swig_paddle.Arguments
paddle/v2/inference.py:        import py_paddle.swig_paddle as api
paddle/v2/parameters.py:        import py_paddle.swig_paddle as api
paddle/v2/parameters.py:        import py_paddle.swig_paddle as api
paddle/v2/parameters.py:    import py_paddle.swig_paddle as api
paddle/v2/__init__.py:    import py_paddle.swig_paddle as api
paddle/v2/event.py:        import py_paddle.swig_paddle as api
paddle/v2/tests/test_parameters.py:    import py_paddle
paddle/v2/tests/test_parameters.py:    del py_paddle
paddle/v2/tests/test_data_feeder.py:import py_paddle.swig_paddle as api
paddle/v2/optimizer.py:        import py_paddle.swig_paddle as swig_api
paddle/v2/optimizer.py:        import py_paddle.swig_paddle as swig_api
paddle/v2/optimizer.py:        import py_paddle.swig_paddle as swig_api
paddle/v2/optimizer.py:        import py_paddle.swig_paddle as swig_api
paddle/v2/optimizer.py:        import py_paddle.swig_paddle as swig_api
paddle/v2/optimizer.py:    import py_paddle.swig_paddle as swig_api
paddle/v2/trainer.py:        import py_paddle.swig_paddle as api
paddle/v2/trainer.py:        import py_paddle.swig_paddle as api
paddle/v2/trainer.py:        import py_paddle.swig_paddle as api
```

**Comment by luotao1** [comment]
@wangkuiyi 
The reason is that `proto/` is not generated in `python/paddle` directory, i.e. we need `/python/paddle/proto`.  I will fix it.

@Xreki 
All these `py_paddle` are in functions, such as:
```
class Optimizer(object):
    def __init__(self, **kwargs):
        import py_paddle.swig_paddle as swig_api
        if 'batch_size' in kwargs:
            del kwargs['batch_size']  # not important for python library.
```
When building the document, it only read `__doc__` of class or function, don't come into themselves.

File: paddle/scripts/travis/build_doc.sh (hunk 2 above)
Comment (luotao1) [inline paddle/scripts/travis/build_doc.sh]: Yes, `build_doc` Travis Job doesn't build twice,  and doesn't build the whole Paddle now.

File: paddle/scripts/travis/build_doc.sh (hunk 1 above)
Comment (luotao1) [inline paddle/scripts/travis/build_doc.sh]: As the next line `git clone $REPO output` , we remain `mkdir output` here.

**Comment by luotao1** [comment]
@Xreki `py_paddle` in `data_feeder.py` will be fixed in next PR. Thx very much for your kindly help.

**Comment by luotao1** [comment]
The travis time of `build_doc` from 35minutes to 5 minutes, and the total travis time is less than 10 minutes.

**Pull request merged**